### PR TITLE
Add distributed query stats

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Relates to https://github.com/kolide/launcher/issues/1147

Adds distributed query stats to results -- schema defined [here](https://github.com/osquery/osquery/blob/master/osquery/distributed/distributed.cpp#L107).

### CODE REVIEW

**Review by James-Pickett — APPROVED** [review]
nice

**Review by seejdev — APPROVED** [review]

**Review by directionless — APPROVED** [review]
Nice, this looks pretty clean

File: plugin/distributed/distributed_test.go [hunk 1]
```diff
@@ -56,16 +56,16 @@ func TestDistributedPlugin(t *testing.T) {
 
 	// Call writeResults
 	getCalled = false
-	resp = plugin.Call(context.Background(), osquery.ExtensionPluginRequest{"action": "writeResults", "results": `{"queries":{"query1":[{"iso_8601":"2017-07-10T22:08:40Z"}],"query2":[{"version":"2.4.0"}]},"statuses":{"query1":"0","query2":"0","query3":"1"}}`})
+	resp = plugin.Call(context.Background(), osquery.ExtensionPluginRequest{"action": "writeResults", "results": `{"queries":{"query1":[{"iso_8601":"2017-07-10T22:08:40Z"}],"query2":[{"version":"2.4.0"}]},"statuses":{"query1":"0","query2":"0","query3":"1"}, "stats":{"query1":{"wall_time_ms": 1, "user_time": 1, "system_time": 1, "memory": 1},"query2":{"wall_time_ms": 2, "user_time": 2, "system_time": 2, "memory": 2},"query3":{"wall_time_ms": 3, "user_time": 3, "system_time": 3, "memory": 3}}}`})
```
[the remark this answers is not in the document]
Comment (RebeccaMahany) [reply plugin/distributed/distributed_test.go:59]: Updated!

**Review by directionless — APPROVED** [review]
I think I'm good. @zwass heads up if this is interesting to y'all.

**Pull request merged**